compose/service.py

```diff
@@ -936,20 +937,55 @@ def build_volume_binding(volume_spec):
     return volume_spec.internal, "{}:{}:{}".format(*volume_spec)
 
 
+def ensure_engine_compatibility(external_path, internal_path):
```

Comment:
Does it matter why we're doing it? I usually keep naming to "what" it's doing, and leave the "why" for the docstring.

To me this is a "normalize" operation not "ensure". "ensure" is validation/assertions, this is "do something in some cases, but ignore it in others".

`normalize_paths_for_engine()` would work for me (even though "engine" might not always be relevant, it's more that we're normalizing them for posix, so my first choice would probably be `normalize_paths_to_posix()` or something like that).